### PR TITLE
blog: Kong Ingress Controller and Istio

### PR DESCRIPTION
**Pull request by devadvocado**
This PR is to add a new blog post on how to combine an ingress controller with a service mesh deployed on Kubernetes. Using Kong Ingress Controller and Istio as open-source options, I wanted to create an easy step-by-step guide for anyone trying to recreate this use case.

### CODE REVIEW

File: content/en/blog/_posts/Kong-Ingress-Controller-and-Service-Mesh.md [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+title: “Kong Ingress Controller and Service Mesh: Setting up Ingress to Istio on Kubernetes”
+date: 2020-02-05
+slug: kong-ingress-controller-and-istio-service-mesh
+---
+
+**Author:** Kevin Chen, Kong
+
+Kubernetes has become the de facto way to orchestrate containers and the services within services. But how do we give services outside our cluster access to what is within? Kubernetes comes with the Ingress API object that manages external access to services within a cluster.
+
+Ingress is a group of rules that will proxy inbound connections to endpoints defined by a backend. However, Kubernetes does not know what to do with Ingress resources without an Ingress controller. This is where an open source controller comes into play. In this post, we are going to use one option for this: the Kong Ingress Controller. The Kong Ingress Controller was open-sourced a year ago and recently reached one million downloads. In the recent 0.7 release, service mesh support was also added. Other features of this release include:
```
Comment (onlydole) [inline content/en/blog/_posts/Kong-Ingress-Controller-and-Service-Mesh.md]:
```suggestion
Ingress is a group of rules that will proxy inbound connections to endpoints defined by a backend. However, Kubernetes does not know what to do with Ingress resources without an Ingress controller, which is where an open source controller can come into play. In this post, we are going to use one option for this: the Kong Ingress Controller. The Kong Ingress Controller was open-sourced a year ago and recently reached one million downloads. In the recent 0.7 release, service mesh support was also added. Other features of this release include:
```

File: content/en/blog/_posts/Kong-Ingress-Controller-and-Service-Mesh.md [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+
+* **Built-In Kubernetes Admission Controller,** which validates Custom Resource Definitions (CRD) as they are created or updated and rejects any invalid configurations. 
+* **n-Memory Mode** - Each pod’s controller actively configures the Kong container in its pod, which limits the blast radius of failure of a single container of Kong or controller container to that pod only.
+* **Native gRPC Routing** - gRPC traffic can now be routed via Kong Ingress Controller natively with support for method-based routing.
+
+![K4K-gRPC](/images/blog/Kong-Ingress-Controller-and-Service-Mesh/KIC-gRPC.png)
+
+If you’d like a deeper dive into Kong Ingress Controller 0.7, please check out the [GitHub repository](https://github.com/Kong/kubernetes-ingress-controller).
```
Comment (onlydole) [inline content/en/blog/_posts/Kong-Ingress-Controller-and-Service-Mesh.md]:
```suggestion
If you would like a deeper dive into Kong Ingress Controller 0.7, please check out the [GitHub repository](https://github.com/Kong/kubernetes-ingress-controller).
```

File: content/en/blog/_posts/Kong-Ingress-Controller-and-Service-Mesh.md [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+details-v1-c5b5f496d-9wm29        2/2     Running   0          101s
+productpage-v1-7d6cfb7dfd-5mc96   2/2     Running   0          100s
+ratings-v1-f745cf57b-hmkwf        2/2     Running   0          101s
+reviews-v1-85c474d9b8-kqcpt       2/2     Running   0          101s
+reviews-v2-ccffdd984-9jnsj        2/2     Running   0          101s
+reviews-v3-98dc67b68-nzw97        2/2     Running   0          101s
+
+```
+
+This command outputs useful data so let’s take a second to understand it. If you examine the READY column, each pod has two containers running: the service and an Envoy sidecar injected alongside it. Another thing to highlight is that there are three review pods but only 1 review service. The Envoy sidecar will load balance the traffic to three different review pods that contain different versions, giving us the ability to A/B test our changes. With that said, you should now be able to access your product page!
```
Comment (onlydole) [inline content/en/blog/_posts/Kong-Ingress-Controller-and-Service-Mesh.md]:
```suggestion
is command outputs useful data, so let’s take a second to understand it. If you examine the READY column, each pod has two containers running: the service and an Envoy sidecar injected alongside it. Another thing to highlight is that there are three review pods but only 1 review service. The Envoy sidecar will load balance the traffic to three different review pods that contain different versions, giving us the ability to A/B test our changes. With that said, you should now be able to access your product page!
```

**Comment by onlydole** [comment]
Hello @devadvocado! Unfortunately, you may have to close this and re-open/re-do the PR due to some issues with Netlify 😢

**Comment by devadvocado** [comment]
hey @onlydole! if you go ahead and close this, I'll create a new PR with the changes requested above 😄  sorry for the slow response, been on the road

**Comment by onlydole** [comment]
No worries at all, @devadvocado! Thank you for the help - I'll close this for now 😄

**Pull request closed without merging**